msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -199,999 +204,1026 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
                         Account = msalAccessTokenCacheItem.HomeAccountId != null ?
                                     new Account(msalAccessTokenCacheItem.HomeAccountId,
                                     preferredUsername, preferredEnvironmentHost) :
-                                    null
```

Comment:
Apollogies for the messy diff, the only changes made here are - whenever the old flag HasStateChanged was set to true: 

- add #pragma warning disable CS0618 // Type or member is obsolete
- make the args flag true